core/admin/port_management.py

```
@@ -0,0 +1,826 @@
+    return {
+        "status": "opened",
+        "service": service_name,
+        "port": external_port,
+        "opened_at": now.isoformat(),
+        "scheduled_close_at": scheduled_close.isoformat(),
+        "opened_by": username,
+        "connection_info": connection_info
+    }
+
+
+def close_monitoring_port_db(db, service_name: str, username: str = None, reason: str = 'manual') -> Dict:
```

Ruff (0.14.11)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)